website/docs/modules/composition.html.markdown

```diff
@@ -106,20 +106,93 @@ module "consul_cluster" {
 }
 ```
 
-This technique is also an answer to the common situation where in one
-configuration a particular object already exists and just needs to be queried,
-while in another configuration the equivalent object must be managed directly.
-This commonly arises, for example, in development environment scenarios where
-certain infrastructure may be shared across development environments for cost
-reasons but managed directly inline in production for isolation.
-
-This is the best way to model such situations in Terraform. We do not recommend
-attempting to construct "if this already exists then use it, otherwise create it"
-conditional configurations; instead, decompose your system into modules and
-decide for each configuration whether each modular object is directly managed or
-merely used by reference. This makes each configuration a more direct description
-of your intent, allowing Terraform to produce a more accurate plan, and making
-it clearer to future maintainers how each configuration is expected to behave.
+### Conditional Creation of Objects
+
+In situations where the same module is used across multiple environments,
+it's common to see that some necessary object already exists in some
+environments but needs to be created in other environments.
+
+For example, this can arise in development environment scenarios: for cost
+reasons, certain infrastructure may be shared across multiple development
+environments, while in production the infrastructure is unique and managed
+directly by the production configuration.
+
+Rather than trying to write a module that itself tries detect whether something
+exists and create it if not, we recommend to apply the dependency inversion
```

Comment:
nitpick, should not block merging, I'm likely mistaken: I think "we recommend applying" is the more common usage (generally only see recommended + infinitive, eg "You are recommended to apply ... ")

Reply:
Yes, this is a good copyedit! I will incorporate it before merging. Thanks!